### PR TITLE
Mapping object to help set baseturfs

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
You place the object in an area.

It sets all the turfs in that area to have the chosen baseturf.

I'm not sure if this is a good idea but the current system of having one substype of every turf for every time you want a different baseturf, or the mapper manually editing baseturfs in every map kinda sucks.

I'll actually test this if a maintainer tells me this solution is allowed.

### CODE REVIEW

**Comment by optimumtact** [comment]
I am not sure about it.

**Comment by KorPhaeron** [comment]
Me neither

**Comment by ChangelingRain** [comment]
Useful but kind of shitty, but useful outweighs shitty for me in this specific case because fuck setting baseturfs.

**Comment by WJohn** [comment]
:+1:

**Comment by xxalpha** [comment]
You call them landmarks but they aren't a landmark subtype and they are never deleted.

Also istypeless loops.

File: code/modules/ruins/objects_and_mobs/ruin_mapping_aids.dm [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+/obj/effect/baseturf_helper
+	name = "lava baseturf editor"
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "syndballoon"
+	var/baseturf = /turf/open/floor/plating/lava/smooth/lava_land_surface
+
+/obj/effect/baseturf_helper/initialize()
+	..()
+	var/area/thearea = get_area(src)
+	for(var/turf/T in get_area_turfs(thearea, z))
```
Comment (xxalpha) [inline code/modules/ruins/objects_and_mobs/ruin_mapping_aids.dm]: This proc not only loops through world, it also returns every turf (in world) under 'thearea' subtypes. Are you sure you want this?

Comment (optimumtact) [inline code/modules/ruins/objects_and_mobs/ruin_mapping_aids.dm]: Do this

Comment (xxalpha) [inline code/modules/ruins/objects_and_mobs/ruin_mapping_aids.dm]: Well this proc is acceptable now, at least it won't be slow. Still, he might not want subtypes of area.

**Pull request merged**